tests/std/tests/P0769R2_shift_left_shift_right/test.cpp

```diff
@@ -7,6 +7,16 @@
 #include <iterator>
 #include <list>
 #include <vector>
+#include <version>

```

Comment:
I think we may consider reverse: adding all tests here. This will avoid adding too many small tests and unnecessarily increase test run time.

Reply:
@AlexGuteniev Thanks for keeping test run time in mind - while it's true that new test directories do increase the total test run time, the added cost is small in comparison to the total cost. I'd generally say that adding a small number of test directories (1 to 3) for a significant new feature is fine. (It's more reasonable for very tiny bugfixes to add very tiny test cases to existing test directories.)

While it would be possible to fuse everything into this test, I think I prefer @MitalAshok's current organization here - the reason is that the new test directories follow the "instantiator" pattern of many existing ranges tests. (For example, I was able to diff the `generate` and `iota` tests which were structurally very similar, and that made things easier to verify and understand.) I think that fusing the tests together would obscure this pattern, and also be somewhat of a distraction from what `P0769R2_shift_left_shift_right` is currently testing.

@miscco Conversely, while it's generally a good idea to keep new test machinery together, I think there may be a compelling argument for what @MitalAshok is doing here - this test `P0769R2_shift_left_shift_right` is focused on runtime behavior (whereas the others are more compile-time focused) and already contains significant setup machinery to test various size and shift values. If this were extracted into another directory or directories, that machinery would need to be replicated, which is somewhat burdensome for maintenance.

So, to summarize: Although it could be done differently, I believe that the current organization is reasonable. Please meow if you still disagree (I am not a ranges expert and you've written a lot more tests recently than I have). :smile_cat: